packages/br-utils/tests/specs/BrUtils.spec.php

```
@@ -0,0 +1,664 @@
+                expect(function () {
+                    new BrUtils(cnpj: ['formatter' => ['dashKey' => "\u{00e5}"]]);
+                })->toThrow(CnpjFormatterOptionsForbiddenKeyCharacterException::class);
+            });
+
+            it('throws CNPJ generator exceptions for invalid generator options', function () {
+                expect(function () {
+                    new BrUtils(cnpj: ['generator' => ['prefix' => '00000000']]);
+                })->toThrow(CnpjGeneratorOptionPrefixInvalidException::class);
+
+                expect(function () {
+                    new BrUtils(cnpj: ['generator' => ['type' => 'invalid']]);
```

PHPStan (2.2.1)

[warning] Parameter $cnpj of class Lacus\BrUtils constructor expects array{formatter?: array{hidden?: bool|null, hiddenKey?: string|null, hiddenStart?: int|null, hiddenEnd?: int|null, dotKey?: string|null, slashKey?: string|null, dashKey?: string|null, escape?: bool|null, ...}|Lacus\BrUtils\Cnpj\CnpjFormatterOptions, generator?: array{format?: bool|null, prefix?: string|null, type?: Lacus\BrUtils\Cnpj\Enums\CnpjType|null}|Lacus\BrUtils\Cnpj\CnpjGeneratorOptions, validator?: array{caseSensitive?: bool|null, type?: 'alphanumeric'|'numeric'|Lacus\BrUtils\Cnpj\Enums\CnpjValidationType|null}|Lacus\BrUtils\Cnpj\CnpjValidatorOptions}|Lacus\BrUtils\Cnpj\CnpjUtils, array{generator: array{type: 'invalid'}} given.
Type `#1` from the union: Offset 'generator' (array{format?: bool|null, prefix?: string|null, type?: Lacus\BrUtils\Cnpj\Enums\CnpjType|null}|Lacus\BrUtils\Cnpj\CnpjGeneratorOptions) does not accept type array{type: 'invalid'}: Type `#1` from the union: Offset 'type' (Lacus\BrUtils\Cnpj\Enums\CnpjType|null) does not accept type string.

(argument.type)